app/stac_api/views.py

```diff
@@ -29,6 +31,55 @@
 logger = logging.getLogger(__name__)
 
 
+def compute_etag(queryset):
+    '''Compute the ETag of the given queryset
+
+    Args:
+        queryset: QuerySet
+            Queryset from which to compute an ETag
+
+    Returns:
+        ETag string
+    '''
+    return hashlib.sha256(''.join([str(i) for i in queryset.values()[0].values()]).encode('utf-8')
+                         ).hexdigest()
```

Comment:
Since the etag is valid for a single object, shouldn't we better pass directly the object itself? also in the methods below we always expect to have only one matching object, wouldn't it be better to directly `.get()` the object?

Second question: are we sure that this doesn't duplicate the DB requests? (once to receive the object in the view function, once to calculate the etag in the decorator)?

Reply:
I could use `get()` instead of `filter()` and enclosed the code in a `try ... except DoesNotExist as err` different approach, don't know if it has different impact on performance ?

You are right this duplicates the DB request, another idea that I had was to compute the ETag in the save as a hidden DB field and to only retrive this field, but I don't know if this has a better performance as retrieving the whole object ? Any other idea on how to compute the ETag ?

Reply:
From this https://www.willmcgugan.com/blog/tech/post/etag-magic-with-django/ we could use the cache (although I would not use a random number, but either the hash of the object content), but not sure if the cache would work with our k8s setup ?

Reply:
After some googling and django docu reading here are my propositions:

1. Use cache as mention in previous comment (add some complexity, don't know how it will work with CloudFront)
1. Use `Collection.objects.get(name=kwargs['collection_name']).only('etag').values('etag')` to retrieve the ETag, this one is then in a hidden DB field and calculated on `save()`. This does a small and quick DB query which only retrieves the Etag
1. Leave it as now (only using the `get()` instead of `filter()`. The use of `values()` is a improvement in performance as the ORM is not used and the fields are not converted in python native)

I suggest using 2., what are your opinions @boecklic @rebert  @hansmannj  ? or any other suggestions ?

Reply:
I'm not in the details. As it is only a lookup and won't impact the performance a lot, I tend to persist the ETag in the DB: 2

Reply:
Without having had too much time to dig into details here, I'd also +1 for option 2. 
Would a re-computation of the ETag also be required and triggered, if for example in case of an asset only the data file on S3 changes, but none of the meta data?

Reply:
@hansmannj  you bring a good point ! Currently I compute the ETag only based on the Model Fields which means that if a Collection.Provider or a Link or an Asset changed the ETag stays the same, which is not what we want ! So solution 3. don't work as we would expect. I will then try to implement solution 2. with taking care about model relations and add unit test for this use case.